packages/ds-core/src/components/button/button.js

```diff
@@ -54,20 +55,23 @@ class Button extends Component {
     this.isComponentMounted = false;
   }
 
-  renderContentWithIcon(text, icon) {
+  renderContentWithIcon(text, icon, isIconOnRightSide) {
+    if (isIconOnRightSide) {
+      return <span><span>{text}</span> <span className={`ds-i right ${icon}`}></span></span>;      
```

Comment:
consider creating renderIcon and renderText methods, to allow changes in one location if needed in the future